TYPO3.Media/Classes/TYPO3/Media/Domain/Model/ThumbnailGenerator/FontDocumentThumbnailGenerator.php

```diff
@@ -0,0 +1,78 @@
+<?php
+namespace TYPO3\Media\Domain\Model\ThumbnailGenerator;
+
+/*
+ * This file is part of the TYPO3.Media package.
+ *
+ * (c) Contributors of the Neos Project - www.neos.io
+ *
+ * This package is Open Source Software. For the full copyright and license
+ * information, please view the LICENSE file which was distributed with this
+ * source code.
+ */
+
+use Imagine\Image\ImagineInterface;
+use TYPO3\Flow\Annotations as Flow;
+use Doctrine\ORM\Mapping as ORM;
+use TYPO3\Flow\Resource\ResourceManager;
+use TYPO3\Media\Domain\Model\Document;
+use TYPO3\Media\Domain\Model\Thumbnail;
+use TYPO3\Media\Exception;
+
+/**
+ * A system-generated preview version of a TTF font
+ */
+class FontDocumentThumbnailGenerator extends AbstractThumbnailGenerator
+{
+
+    /**
+     * @param Thumbnail $thumbnail
+     * @return boolean
+     */
+    public function canRefresh(Thumbnail $thumbnail)
+    {
+        return (
+            $this->isExtensionSupported($thumbnail) &&
+            function_exists('imagecreatetruecolor')
+        );
+    }
+
+    /**
+     * @param Thumbnail $thumbnail
+     * @return void
+     * @throws Exception\NoThumbnailAvailableException
+     */
+    public function refresh(Thumbnail $thumbnail)
+    {
+        $temporaryPathAndFilename = null;
+        try {
+            $filename = pathinfo($thumbnail->getOriginalAsset()->getResource()->getFilename(), PATHINFO_FILENAME);
+
+            $temporaryLocalCopyFilename = $thumbnail->getOriginalAsset()->getResource()->createTemporaryLocalCopy();
+            $temporaryPathAndFilename = $this->environment->getPathToTemporaryDirectory() . uniqid('ProcessedFontThumbnail-') . '.' . $filename . '.jpg';
+
+            $im = imagecreatetruecolor(500, 500);
+            $red = imagecolorallocate($im, 0xFF, 0xFF, 0xFF);
+            $black = imagecolorallocate($im, 0x00, 0x00, 0x00);
+
+            imagefilledrectangle($im, 0, 0, 500, 500, $red);
+            imagefttext($im, 24, 0, 40, 50, $black, $temporaryLocalCopyFilename, 'Neos Font Preview');
+            imagefttext($im, 16, 0, 40, 85, $black, $temporaryLocalCopyFilename, 'ABCDEFGHIJK');
+            imagefttext($im, 16, 0, 40, 115, $black, $temporaryLocalCopyFilename, 'abcdefghijk');
+            imagefttext($im, 16, 0, 40, 145, $black, $temporaryLocalCopyFilename, '1234567890');
+            imagefttext($im, 16, 0, 40, 175, $black, $temporaryLocalCopyFilename, 'If something can corrupt you,');
+            imagefttext($im, 16, 0, 40, 200, $black, $temporaryLocalCopyFilename, 'you\'re corrupted already.');
+
+            imagejpeg($im, $temporaryPathAndFilename);
+
+            $resource = $this->resourceManager->importResource($temporaryPathAndFilename);
+            $thumbnail->setResource($resource);
+
+            $this->unlinkTemporaryFiles([$temporaryPathAndFilename]);
+        } catch (\Exception $exception) {
+            $this->unlinkTemporaryFiles([$temporaryPathAndFilename]);
```

Comment:
would $temporaryPathAndFilename actually contain anything if an exception happened in that block?